From david-b@pacbell.net Mon Nov  7 15:43:21 2005
Message-ID: <20051107233824.60767.qmail@web80708.mail.yahoo.com>
Date: Mon, 7 Nov 2005 15:38:24 -0800 (PST)
From: David Brownell <david-b@pacbell.net>
Subject: USB: wakeup flag updates (3/3) isp116x-hcd
To: greg@kroah.com
Cc: linux-usb-devel@lists.sourceforge.net


This makes the ISP116x HCD use the driver model wakeup flags for its
controller, not the flags in the HCD glue (which will be removed).

Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/isp116x-hcd.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/usb/host/isp116x-hcd.c
+++ gregkh-2.6/drivers/usb/host/isp116x-hcd.c
@@ -1378,7 +1378,8 @@ static int isp116x_start(struct usb_hcd 
 
 	val = 0;
 	if (board->remote_wakeup_enable) {
-		hcd->can_wakeup = 1;
+		if (!device_can_wakeup(hcd->self.controller))
+			device_init_wakeup(hcd->self.controller, 1);
 		val |= RH_HS_DRWE;
 	}
 	isp116x_write_reg32(isp116x, HCRHSTATUS, val);
@@ -1428,7 +1429,7 @@ static int isp116x_bus_suspend(struct us
 		hcd->state = HC_STATE_QUIESCING;
 		val &= (~HCCONTROL_HCFS & ~HCCONTROL_RWE);
 		val |= HCCONTROL_USB_SUSPEND;
-		if (hcd->remote_wakeup)
+		if (device_may_wakeup(&hcd->self.root_hub->dev))
 			val |= HCCONTROL_RWE;
 		/* Wait for usb transfers to finish */
 		mdelay(2);
